src/nats/mod.rs

```diff
@@ -0,0 +1,243 @@
+use std::borrow::Cow;
+
+use testcontainers::{core::WaitFor, Image, RunnableImage};
+
+/// Nats image for [testcontainers](https://crates.io/crates/testcontainers).
+///
+/// This image is based on the official [Nats](https://hub.docker.com/_/nats) image.
+/// The default user is `` and the default password is ``.
+
+#[derive(Clone, Debug, PartialEq, Eq)]
+pub struct Nats {
+    version: Value,
+    user: Option<Value>,
+    pass: Option<Value>,
+}
+
+impl Nats {
+    const DEFAULT_USER: &'static str = "";
+    const DEFAULT_PASS: &'static str = "";
+    const DEFAULT_VERSION_TAG: &'static str = "2.10.14";
+
+    /// Create a new instance of a Nats image.
+    #[must_use]
+    pub fn new() -> Self {
+        Self {
+            version: Cow::Borrowed(Self::DEFAULT_VERSION_TAG),
+            user: Some(Cow::Borrowed(Self::DEFAULT_USER)),
+            pass: Some(Cow::Borrowed(Self::DEFAULT_PASS)),
+        }
+    }
+
+    /// Set the Nats version to use.
+    /// The value must be an existing Nats version tag.
+    pub fn with_version(mut self, version: impl Into<Value>) -> Self {
```

Comment:
We usually don't expose such methods as part of `Image` implementation. It's possible to override the tag with `RunnableImage`

Reply:
[26daf99](https://github.com/testcontainers/testcontainers-rs-modules-community/pull/123/commits/26daf99273e874cc6b792256e793301a0e56e198)